src/assets/scripts/client/navigationLibrary/StandardRoute/StandardRouteWaypointModel.js

```diff
@@ -283,23 +283,31 @@ export default class StandardRouteWaypointModel extends BaseModel {
      * @for StandardRouteWaypointModel
      * @method clonePositionFromFix
      * @param fixCollection {FixCollection}
+     * @return {StandardRouteWaypointModel}
      * @private
+     * @chainable
      */
     clonePositionFromFix() {
-        if (this.name[0] === VECTOR_WAYPOINT_PREFIX) {
+        const isFlyOverWaypoint = this.name.indexOf(FLY_OVER_WAYPOINT_PREFIX) !== -1;
+        const isHoldWaypoint = this.name.indexOf(HOLD_WAYPOINT_PREFIX) !== -1;
+        const isVectorWaypoint = this.name.indexOf(VECTOR_WAYPOINT_PREFIX) !== -1;
```

Comment:
Hm, you have a point. Although actually no, it was the `WaypointModel` that has the handy methods. All the standard route stuff used to kinda make sense to me, though with time it's feeling more and more like a pretty confusing maze of things. I think you're right, we could use better names, and I _feel_ like we may be able to simplify. But you have spent much more time in that area than me, so perhaps not. Regardless, I can access `this._isVector` and `this._isFlyOverWaypoint`, but I don't have anything else like that for holds. I could delve into figuring it out, but at this point it doesn't seem like that'd quite be worth it in my opinion.